src/modules/map/components/openlayers/utils/print-area-renderer.composable.js

```diff
@@ -0,0 +1,207 @@
+            olHas.DEVICE_PIXEL_RATIO
+        var h =
+            (((((size.height / POINTS_PER_INCH) * MM_PER_INCHES) / 1000.0) * s) / resolution) *
+            olHas.DEVICE_PIXEL_RATIO
+        var mapSize = map.getSize()
+        var center = [
+            (mapSize[0] * olHas.DEVICE_PIXEL_RATIO) / 2,
+            (mapSize[1] * olHas.DEVICE_PIXEL_RATIO) / 2,
+        ]
+
+        var minx, miny, maxx, maxy
+
+        minx = center[0] - w / 2
+        miny = center[1] - h / 2
+        maxx = center[0] + w / 2
+        maxy = center[1] + h / 2
+        return [minx, miny, maxx, maxy]
+    }
+
+    // Compose events
+    function handlePreRender(event) {
+        var context = event.context
+        context.save()
+    }
+    function handlePostRender(event) {
+        // This is where we draw the print area rectangle using the worldPolygon
+        var context = event.context
+        const size = map.getSize()
+
+        var height = size[1] * olHas.DEVICE_PIXEL_RATIO,
+            width = size[0] * olHas.DEVICE_PIXEL_RATIO
+
+        var minx = printRectangle[0],
+            miny = printRectangle[1],
+            maxx = printRectangle[2],
+            maxy = printRectangle[3]
+
+        context.save()
+
+        context.beginPath()
+        // Outside polygon, must be clockwise
+        context.moveTo(0, 0)
+        context.lineTo(width, 0)
+        context.lineTo(width, height)
+        context.lineTo(0, height)
+        context.lineTo(0, 0)
+        context.closePath()
+
+        // Inner polygon,must be counter-clockwise
+        context.moveTo(minx, miny)
+        context.lineTo(minx, maxy)
+        context.lineTo(maxx, maxy)
+        context.lineTo(maxx, miny)
+        context.lineTo(minx, miny)
+        context.closePath()
+
+        context.fillStyle = 'rgba(0, 5, 25, 0.75)'
+        context.fill()
+
+        context.restore()
+    }
+
+    // Compute the optimal scale based on the map size (layout), resolution,
+    // non-covered map vie (by header and menu tray)
+    function getOptimalScale() {
+        var size = map.getSize()
+        var resolution = map.getView().getResolution()
+        var widthMargin = store.state.ui.menuTrayWidth
+        var heightMargin = store.state.ui.headerHeight
+        var width = resolution * (size[0] - widthMargin * 2)
+        var height = resolution * (size[1] - heightMargin * 2)
+        var layoutSize = store.getters.mapSize
+        var scaleWidth = (width * UNITS_RATIO * POINTS_PER_INCH) / layoutSize.width
+        var scaleHeight = (height * UNITS_RATIO * POINTS_PER_INCH) / layoutSize.height
+        var testScale = scaleWidth
+        if (scaleHeight < testScale) {
+            testScale = scaleHeight
+        }
+        var nextBiggest = null
+        var selectedLayoutScales = Array.from(store.state.print.selectedLayout.scales)
+        // Make sure it's sorted ascending
+        selectedLayoutScales.sort((a, b) => a - b)
+        selectedLayoutScales.forEach(function (scale) {
```

Comment:
what you're doing here is about the same as `selectedLayoutScales.find((scale) => scale < testScale)` which will give you the first entry to match your predicate

Reply:
I need to sort descending with that code, but that's ok, I use it, since it's simpler